examples/sample/release.yaml

```diff
@@ -31,5 +31,28 @@ spec:
 #         name: svals
 #         namespace: wordpress
 #         optional: false
+#  connectionDetails:
```

Comment:
I feel connectionDetails isn't the perfect name as it's really general purpose data sync from a Release.

releaseObservation:
 ...
  writeObservationSecretToRef:

Reply:
I see what you mean and actually give it a try to change it in that direction, however ended up with current solution because of following reasons:

- `writeConnectionSecretToRef` is part of [crossplane managed resource spec](https://github.com/crossplane/crossplane-runtime/blob/master/apis/common/v1/resource.go#L137), there are existing machinery coming with crossplane runtime hence this field is common across all providers. I preferred to reuse this pattern/machinery here.
- `connectionDetails` is also something crossplane users should be familiar with, it exists in composite resource api ([example](https://github.com/upbound/platform-ref-aws/blob/master/cluster/composition.yaml#L14)) with similar purposes, where you select what you want to be included in the connection secret that composite resource creates.